packages/backend/src/index.ts

```diff
@@ -168,6 +169,17 @@ async function main() {
 
   const apiRouter = Router();
 
+  // Scalprum frontend plugins provider
+  const scalprumEmv = useHotMemoize(module, () => createEnv('scalprum'));
+  apiRouter.use(
+    '/scalprum',
+    await scalprumRouter({
+      logger: scalprumEmv.logger,
+      pluginManager,
+      discovery: scalprumEmv.discovery,
+    }),
+  );
```

Comment:
can't you use `addPlugin` factory?

Reply:
No, because the `createRouter` method expected in `addPlugin` takes a `PluginEnvironment` as an argument.
But the `LegacyPluginEnvironment` provides a `PluginProvider` interface, and not the `PluginManager` class.
But in this case we need the `PluginManager (to get the frontend plugins, for example).
I assume that the PluginProvider should be completed on upstream, but in the meantime we can do this way.